### PR TITLE
Yield self

### PR DESCRIPTION
**Pull request by roback**
Nice to have. Added it for the following methods:

* `Client.new`
* `Client#query` (and `Query.new`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,9 +25,11 @@ gem 'twingly-search'
 require 'twingly/search'
 
 client = Twingly::Search::Client.new
```
Comment (jage) [inline README.md]: How about we show how to set the user agent here? Something like:

```Ruby
client = Twingly::Search::Client.new do |client|
  client.user_agent = "MyCompany/1.0"
end
```

I'm thinking we need to promote it if we want our customers to set a custom user agent.

**Comment by jage** [comment]
LGTM!

File: README.md (hunk 1 above)
Comment (jage) [inline README.md]: > I'm thinking we need to promote it if we want our customers to set a custom user agent.

Hmm, do we want our customers to do this? It's nice to see which of our clients/versions they are using.

Comment (walro) [inline README.md]: *late to the part* we could append their own user agent and always include the part we want

Comment (roback) [inline README.md]: I knew there was something I forgot :)

**Pull request merged**